File: mpikurs_solutions/f/hello_mpi_r.f
      program hello
      implicit none
      include 'mpif.h'
      integer nproc, myid, version, subversion, ierr, pnamelen
      character*(40) pname
c---------------------------------------------------------- 
c        start MPI
c----------------------------------------------------------
      call MPI_INIT( ierr )
      call MPI_COMM_SIZE( MPI_COMM_WORLD, nproc, ierr )
      call MPI_COMM_RANK( MPI_COMM_WORLD, myid, ierr )
      call MPI_GET_PROCESSOR_NAME( pname, pnamelen, ierr ) 
      write(6,*)nproc, myid, pname
      if (myid.eq.0) then
        call MPI_GET_VERSION( version, subversion, ierr )
        write(6,*) 'version: ',version,'subversion:',subversion
      end if

      call MPI_FINALIZE (ierr)

      stop
      end

File: mpikurs_solutions/c/hello_mpi_r.c
#include<mpi.h>
#include<stdio.h>

int main(int argc,char **argv)
{
  int np, me, resultlen;
  char name[41];

  MPI_Init(&argc,&argv);
  MPI_Comm_size(MPI_COMM_WORLD,&np);
  MPI_Comm_rank(MPI_COMM_WORLD,&me);
  MPI_Get_processor_name(name, &resultlen);

  printf("hello %i %i %s \n",np,me,name);

  if (me==0)
  { int version, subversion;
    MPI_Get_version(&version, &subversion);
    printf("MPI version %i.%i \n",version, subversion);
  }

  MPI_Finalize();
  return 0;
}
